agent/service.go

```diff
@@ -185,9 +187,14 @@ func (as *agentService) Data(ctx context.Context, dataset Dataset) error {
 	if hash != as.computation.Datasets[index].Hash {
 		return ErrHashMismatch
 	}
+
+	if as.computation.Datasets[index].Filename != "" && as.computation.Datasets[index].Filename != dataset.Filename {
+		return ErrFileNameMismatch
+	}
+
 	as.computation.Datasets = slices.Delete(as.computation.Datasets, index, index+1)
 
-	f, err := os.Create(fmt.Sprintf("%s/dataset-%d", algorithm.DatasetsDir, index))
+	f, err := os.Create(fmt.Sprintf("%s/%s", algorithm.DatasetsDir, dataset.Filename))
```

Comment:
Handle an instance where the filename is empty so that you do not have a dataset without a name.

Reply:
i.e., if dataset is empty, save the dataset as dataset-1 and so on.

Reply:
Since file is uploaded from cli, the filename can not be empty so this approach is ok.